.github/workflows/go.yml

```diff
@@ -39,6 +43,15 @@ jobs:
   
     - name: Build
       run: go build .
-
+       
+    - name: Format  
+      run: go fmt
+    
     - name: Test
       run: go test -v
+  
+    - name: Vet 
+      run: go vet -v
+    
+    - name: Coverage
+      run: go test -v -covermode=atomic
```

Comment:
We can generate a coverage file from this command, we should create an issue to add this later.

Reply:
```suggestion
      run: go test -v -covermode=atomic -coverprofile=coverage.tx ./... && go tool cover -func=coverage.tx
```